frontend/src/hocs/withHostCommunication/withHostCommunication.tsx

```diff
@@ -71,7 +92,21 @@ function withHostCommunication(
     const [deployedAppMetadata, setDeployedAppMetadata] = useState({})
     const [toolbarItems, setToolbarItems] = useState<IToolbarItem[]>([])
 
+    const [allowedOriginsResp, setAllowedOriginsResp] =
+      useState<IAllowedMessageOriginsResponse | null>(null)
+    const [deferredAuthToken, setDeferredAuthToken] =
+      useState<DeferredValue>(createDeferredValue)
+
     useEffect(() => {
+      if (!allowedOriginsResp) {
+        return () => {}
+      }
+
+      const { allowedOrigins, useExternalAuthToken } = allowedOriginsResp
+      if (!useExternalAuthToken) {
+        deferredAuthToken.resolve()
+      }
+
```

Comment:
Not a criticism of your PR, but this whole class is sooooo hairy. It feels essentially write-only at this point. 

I feel like we should prioritize a `withHostCommunication` refactor in the near-near future. (Should it even be a React component?)

Reply:
Yeah, agreed. I'd support seeing if we can get this made an official Q1 goal given that this code is changing far more frequently than I would have expected it to when I first noticed it was pretty hairy